### PR TITLE
Add test cases for JSON resource marshaling

### PR DESCRIPTION
**Pull request by sagar23sj**
Resources Covered :

- ServiceHook
- CodeOfConduct
- markdownRequest

Helps with issue: #55

### CODE REVIEW

File: github/misc_test.go [hunk 1]
```diff
@@ -348,3 +348,68 @@ func TestListServiceHooks(t *testing.T) {
 		return resp, err
 	})
 }
+
+func TestMarkdownRequest_marshal(t *testing.T) {
```
Comment (gmlewis) [inline github/misc_test.go:352]:
```suggestion
func TestMarkdownRequest_Marshal(t *testing.T) {
```

File: github/misc_test.go [hunk 2]
```diff
@@ -348,3 +348,68 @@ func TestListServiceHooks(t *testing.T) {
...
+	want := `{
+		"text": "txt",
+		"mode": "mode",
+		"context": "ctx"
+	}`
+
+	testJSONMarshal(t, a, want)
+}
+
+func TestCodeOfConduct_marshal(t *testing.T) {
```
Comment (gmlewis) [inline github/misc_test.go:370]:
```suggestion
func TestCodeOfConduct_Marshal(t *testing.T) {
```

File: github/misc_test.go [hunk 3]
```diff
@@ -348,3 +348,68 @@ func TestListServiceHooks(t *testing.T) {
...
+		"name": "name",
+		"key": "key",
+		"url": "url",
+		"body": "body"
+	}`
+
+	testJSONMarshal(t, a, want)
+}
+
+func TestServiceHook_marshal(t *testing.T) {
```
Comment (gmlewis) [inline github/misc_test.go:390]:
```suggestion
func TestServiceHook_Marshal(t *testing.T) {
```

**Review by gmlewis — CHANGES_REQUESTED** [review]

**Review by gmlewis — APPROVED** [review]
Thank you, @sagar23sj !
LGTM.

Awaiting second LGTM before merging.

**Review by Parker77 — APPROVED** [review]
LGTM.

**Pull request merged**